astropy/units/core.py

```diff
@@ -2072,7 +2059,7 @@ def __call__(
         if isinstance(s, UnitBase):
             return s
 
-        elif isinstance(s, (bytes, str)):
+        if isinstance(s, (str, bytes)):
```

Comment:
I was going to ask to leave the `elif` in place, even though there are `return` statements in every branch, as I find it much easier to see that way that they are all different checks on `s`. 

But given my note about that the `Quantity` part can go down, this is better, as that (and `Complex`) needs a local import.

p.s. Switch of `str` and `bytes` is nice; `bytes` is far less likely.